src/main/java/de/tum/in/www1/artemis/service/user/UserCreationService.java

```diff
@@ -127,7 +96,8 @@ public User createUser(String login, @Nullable String password, Set<String> grou
         newUser.setLogin(login);
         newUser.setFirstName(firstName);
         newUser.setLastName(lastName);
-        newUser.setGroups(groups);
+        // needs to be mutable --> new HashSet<>(Set.of())
+        newUser.setGroups(groups != null ? groups : new HashSet<>(Set.of()));
```

Comment:
```suggestion
        newUser.setGroups(groups != null ? new HashSet<>(groups) : new HashSet<>());
```
This makes sure that the Set is always mutable. Also Set.of() is not needed to create an empty Set.